core/src/main/java/amino/run/kernel/server/KernelServer.java

```diff
@@ -29,6 +32,29 @@ void copyKernelObject(KernelOID oid, KernelObject object)
                     KernelObjectStubNotCreatedException, MicroServiceNotFoundException,
                     MicroServiceReplicaNotFoundException;
 
+    /**
+     * Updates a kernel server about the list of all available kernel servers managed by the OMS
+     *
+     * @param serverList
+     * @throws RemoteException
+     */
+    void updateAvailableKernelServers(List<InetSocketAddress> serverList) throws RemoteException;
+
+    /**
+     * Receive HeartBeat with random bytes to measure data transfer time
+     *
+     * @param data
+     * @throws RemoteException
+     */
+    void receiveHeartBeat(RandomData data) throws RemoteException;
```

Comment:
It would be good to add comments explaining why it doesn't just take a String as a parameter. Superficially it looks quite silly to create a whole new data type just for this purpose.  I understand you're probably trying to make the serialization and deserialization more efficient or something, but I think there's probably a much better way (e.g. by using a String.subString() on the fly to pass in.

Reply:
Added comments for RandomData class.